### ISSUE TITLE
UninitializedPropertyAccessException on setPushToken

### ISSUE THREAD

**Issue by KamilSucharski** [klaviyo/klaviyo-android-sdk#116]
### Description

```markdown
When setting a push token the app might crash. It's a rare crash that I can't recreate, but it does happen for some people.
```

### Checklist

- [ ] I have determined whether this bug is also reproducible in a vanilla Android project
- [ ] If possible, I've reproduced the issue using the `main` branch of this package.
- [X] This issue hasn't been addressed in an [existing GitHub issue](https://github.com/klaviyo/klaviyo-android-sdk/issues) or [pull request](https://github.com/klaviyo/klaviyo-android-sdk/pulls).

### Expected behavior

```markdown
Push token should be set (and it is most of the time!).
```

### Actual behavior

```markdown
Fatal Exception: kotlin.UninitializedPropertyAccessException: lateinit property initialBody has not been initialized
       at com.klaviyo.analytics.networking.requests.PushTokenApiRequest.equals(PushTokenApiRequest.kt:96)
       at java.util.concurrent.ConcurrentLinkedDeque.contains(ConcurrentLinkedDeque.java:1081)
       at com.klaviyo.analytics.networking.KlaviyoApiClient.enqueueRequest(KlaviyoApiClient.kt:118)
       at com.klaviyo.analytics.networking.KlaviyoApiClient.enqueuePushToken(KlaviyoApiClient.kt:81)
       at com.klaviyo.analytics.Klaviyo.setPushToken(Klaviyo.kt:153)
```
```


### Steps to reproduce

```markdown
Klaviyo.initialize(BuildConfig.KLAVIYO_API_KEY, application)
application.registerActivityLifecycleCallbacks(Klaviyo.lifecycleCallbacks)
FirebaseMessaging.getInstance().token.addOnSuccessListener { pushToken ->
    Klaviyo.setPushToken(pushToken)
}
```

### The Klaviyo Android SDK version information

1.3.5

### Device Information

Samsung Galaxy S21 FE 5G, Android 14 | Samsung Galaxy Note9, Android 10 | Samsung Galaxy S9+, Android 10 | Samsung Galaxy A42 5G, Android 12

### Android Studio Version

2023.1.1

### Android API Level

34

**Comment by jordan-griffin** [comment]
Hey, thanks for reporting. Though we have failed to reproduce on our end either - we're looking into this and have a few ideas on how we might remediate the issue

**Comment by evan-masseau** [comment]
I've identified a repro and will have a patch out soon. This occurs when a push request is decoded from JSON and then compared with the equality operator we recently added for de-duping logic.

**Comment by evan-masseau** [comment]
Thanks again, we released 1.4.1 to patch the bug!
